framework/src/base/FEProblem.C

```diff
@@ -1494,6 +1491,15 @@ FEProblem::addAuxKernel(const std::string & kernel_name, const std::string & nam
     parameters.set<SystemBase *>("_sys") = &_aux;
     parameters.set<SystemBase *>("_nl_sys") = &_nl;
   }
+
+  if (_displaced_problem != NULL)
+  {
+    if (!parameters.get<std::vector<BoundaryName> >("boundary").empty())
+      _reinit_displaced_face = true;
```

Comment:
This is REALLY not good... could be REALLY expensive.  We only want to do this if we really have to.